File: Frist DataStructure (Arrays)/Java/BinarySearch.java

public class BinarySearch {

 public static int binarySearch(int arr[], int n,int k){
    int start=0;
    int end=n;
    while (start<=end) {
        int mid= (start+end)/2;
        if(k==arr[mid]){
            return mid;
        }if (mid<k) {
            start=mid+1;
        } else {
            end=mid-1;
        }

        start++;
        end--;

        
    }
    return -1;
 }
  public static void main(String[] args) {
    int arr[]={1,2,3,4,5,6,7,8,9,10};
    int n=arr.length;
    int k=20;
    int ans = binarySearch(arr,n,k);
    if(ans==-1)
    System.out.println("Element is not found");
    else
    System.out.println("Element present at index :"+ ans );
  }
    
}

File: Frist DataStructure (Arrays)/JavaScript/BinarySearch.js
  function BinarySearch(arr,n,k){
    let start=0; 
    let end=n-1;
    

    while(start<=end){
        let mid = Math.floor((start + end) / 2)

        if(arr[mid]==k){
            return mid

        } else if (arr[mid]<k) {
            start=mid+1
        } else {
            end=mid-1
        }
    }
    return -1;
  }



   arr=new Array(1,2,3,4,5,6,7,8,9,10)
   n=arr.length
   k=8

   ans= BinarySearch(arr,n,k)

  if(ans==-1){
    console.log("Element is not found")
  }else{
    console.log("Element id present at index :" +ans)
  }